### DISCUSSION TITLE
Implementing EIP-6492 along with ERC-1271?

### DISCUSSION THREAD

**Discussion by peterferguson** [wevm/viem#337] [Idea / Feature Request] [answered]
Hey I seen that work has begun on 1271 implementation #320 & just thought I would drop a message about [eip-6492](https://eips.ethereum.org/EIPS/eip-6492).

It is a backwards compatible extension to 1271 that allows pre-deployed contracts to sign messages. Providing pretty huge UX gains for contract wallets (allowing them to not be deployed until first transaction while still being able to 'sign-in' to dApps). 

The eip is early but there are issues in [siwe](https://github.com/spruceid/siwe/issues/148) & I thought I saw one to ethers but it seems to have been deleted 🤷.

I would be happy to help out with the implementation also if you think it is something that should be implemented in viem 👍

**Comment by tmm** [comment]
Thanks for raising! Let's see how the EIP progresses before adding in support. If you are following that conversation, feel free to report back here when things seem further along and we can start figuring out what support would look like in viem.

**Comment by Ivshti** [comment]
@tmm I'd say the EIP is stable at this point after passing multiple stages of discussion.

**Reply by jxom** [reply, marked as answer]
viem now implements 6492 😊

**Reply by Ivshti** [reply]
that's fantastic news!
